src/services/services.ts

```diff
@@ -345,6 +346,88 @@ namespace ts {
         }
     }
 
+    class TokenOrIdentifierObject implements Token {
+        public kind: SyntaxKind;
+        public flags: NodeFlags;
+        public parent: Node;
+        public jsDocComments: JSDocComment[];
+        public __tokenTag: any;
+
+        constructor(public pos: number, public end: number) {

```

Comment:
i understand. it just took me a bit to verify that by looking at the ones done in the constructor explicitly and the ones done because they are parameter properties.

my comment is to not use parameter properties just to make it clear what the order is.

also worth adding a comment that these need to stay in sync on all three classes.